File: CodeForces/1301-1400/1335/A/1335A.swift
// Problem 1335A: Candies and Two Sisters
// https://codeforces.com/contest/1335/problem/A
// Executed with: Swift version 6.2 (swift-6.2-RELEASE)
// Linux version: Linux kernel 6.17.0-5-generic
// Written on: October 26th, 2025
// Codeforces language used: Cannot be submitted

import Foundation

var runs = Int(readLine()!)!

for _ in 0...runs-1 {
    let num = Double(readLine()!)!
    print(Int(floor((num-1.0)/2.0)))
}

File: CodeForces/1301-1400/1335/A/1335A.cs
// Problem 1335A: Candies and Two Sisters
// https://codeforces.com/contest/1335/problem/A
// Compiled with: Mono JIT compiler version 6.12.0.199 (tarball Fri May  2 12:22:51 UTC 2025)
// Linux version: Linux kernel 6.17.0-5-generic
// Submitted on: October 26th, 2025
// Codeforces language used: C# Mono 6.8

using System;

class Program {
    static void Main() {
        int runs = int.Parse(Console.ReadLine());
        for (int i = 0; i < runs; i++){
            int candy = int.Parse(Console.ReadLine());
            Console.WriteLine(Math.Floor((candy-1.0) / 2.0));
        }
    }
}
